nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-properties-loader/src/main/java/org/apache/nifi/properties/NiFiPropertiesLoader.java

```diff
@@ -174,6 +177,19 @@ ProtectedNiFiProperties readProtectedPropertiesFromDisk(File file) {
             rawProperties.load(inStream);
             logger.info("Loaded {} properties from {}", rawProperties.size(), file.getAbsolutePath());
 
+            // Trim whitespace from each property. If property is multi-line, remove anything after the first line break.
+            Set<String> keys = rawProperties.stringPropertyNames();
+            Iterator<String> itr = keys.iterator();
+            final Pattern whitespaceRegex = Pattern.compile("\\s+$");
+            while(itr.hasNext()){
+                String key = itr.next();
+                String prop = rawProperties.getProperty(key);
+                if(!prop.isEmpty()){
```

Comment:
Did you consider using Commons Lang 3 `StringUtils.stripEnd()`?  That dependency is already part of `nifi-properties-loader`, and it handles null values, so it would remove the need for the conditional as well as the regular expression pattern.

Reply:
I'm not really familiar with the Apache Commons library (yet). I've made the suggested change.